Username: testRedditAPI2021
Password: testAPI2021


In [ ]:
pip install praw 

     |████████████████████████████████| 163kB 6.7MB/s 
     |████████████████████████████████| 71kB 5.4MB/s 


In [ ]:
import praw
import pandas as pd

In [ ]:
reddit = praw.Reddit(client_id='ADwwSL9jTbAeeQ', client_secret='c5Iib_DtxXFKziu6JPdJnYjsI1ZwyQ', user_agent='Depression Detection')
depression = reddit.subreddit('depression')
depressed = reddit.subreddit('depressed')
nondepression = reddit.subreddit('CasualConversation')
nondepression2 = reddit.subreddit('casual')
nondepression3 = reddit.subreddit('chat')

In [ ]:
x = next(depression.top())
dir(x)

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.



['STR_FIELD',
 '__class__',
 '__delattr__',
 '__dict__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattr__',
 '__getattribute__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__le__',
 '__lt__',
 '__module__',
 '__ne__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 '__weakref__',
 '_chunk',
 '_comments_by_id',
 '_fetch',
 '_fetch_data',
 '_fetch_info',
 '_fetched',
 '_kind',
 '_reddit',
 '_reset_attributes',
 '_safely_add_arguments',
 '_url_parts',
 '_vote',
 'all_awardings',
 'allow_live_comments',
 'approved_at_utc',
 'approved_by',
 'archived',
 'author',
 'author_flair_background_color',
 'author_flair_css_class',
 'author_flair_richtext',
 'author_flair_template_id',
 'author_flair_text',
 'author_flair_text_color',
 'author_flair_type',
 'author_fullname',
 'author_patreon_flair',
 'author_premium',
 'award',
 'awarders',
 'banned_at_utc',
 'banned_by',
 'can_gil

In [ ]:
print(x.selftext)


It’s a constant sense of wanting to go somewhere else. I’d feel better if i was at this place. Then you go there and then it’s the same exact feeling, just in a different physical location. 

I feel mentally homeless and I just want to go home.


Edit: I didn’t think this would get traction and that’s my first gold. Thanks buddy.


In [ ]:
depressed_posts = []
for post in depression.top(limit=1000000):
     depressed_posts.append([post.author, post.selftext, 1])
print(len(depressed_posts))
for post in depression.hot(limit=1000000):
     depressed_posts.append([post.author, post.selftext, 1])
print(len(depressed_posts))
for post in depressed.top(limit=1000000):
     depressed_posts.append([post.author, post.selftext, 1])
print(len(depressed_posts))
for post in depressed.hot(limit=1000000):
     depressed_posts.append([post.author, post.selftext, 1])
print(len(depressed_posts))

depressed_all = set(tuple(i) for i in depressed_posts)
print(len(depressed_all))

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readt

969


It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readt

1929


It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readt

2928


It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readt

3852
3736


In [ ]:
nondepressed_posts = []
for post in nondepression.top(limit=1000000):
     nondepressed_posts.append([post.author, post.selftext, 0])
print(len(nondepressed_posts))
for post in nondepression.hot(limit=1000000):
     nondepressed_posts.append([post.author, post.selftext, 0])
print(len(nondepressed_posts))
for post in nondepression2.top(limit=1000000):
     nondepressed_posts.append([post.author, post.selftext, 0])
print(len(nondepressed_posts))
for post in nondepression2.hot(limit=1000000):
     nondepressed_posts.append([post.author, post.selftext, 0])
print(len(nondepressed_posts))
for post in nondepression3.top(limit=1000000):
     nondepressed_posts.append([post.author, post.selftext, 0])
print(len(nondepressed_posts))
for post in nondepression3.hot(limit=1000000):
     nondepressed_posts.append([post.author, post.selftext, 0])
print(len(nondepressed_posts))

nondepressed_all = set(tuple(i) for i in nondepressed_posts)
print(len(nondepressed_all))

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readt

989


It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readt

1779


It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.



1836


It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.



1893


It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readt

2891


It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readthedocs.io.
See https://praw.readthedocs.io/en/latest/getting_started/multiple_instances.html#discord-bots-and-asynchronous-environments for more info.

It appears that you are using PRAW in an asynchronous environment.
It is strongly recommended to use Async PRAW: https://asyncpraw.readt

3752
3516


=> We found 3736 data entries for depressed texts and 3516 entries for non-depressed text. This would make up a fairly balanced dataset.

In [ ]:
depressed_df = pd.DataFrame(depressed_all,columns=['author', 'post', 'label'])
nondepressed_df = pd.DataFrame(nondepressed_all,columns=['author', 'post', 'label'])

In [ ]:
print("___________DEPRESSED_____________")
print(depressed_df)
print("___________NOT DEPRESSED_____________")
print(nondepressed_df)

___________DEPRESSED_____________
                    author  ... label
0         theky0t0pr0t0c0l  ...     1
1         Coronaisgayforme  ...     1
2       EggplantSpecialist  ...     1
3     2realtheartistdotcom  ...     1
4               vargas_98e  ...     1
...                    ...  ...   ...
3731          slicehishand  ...     1
3732            hortiz1598  ...     1
3733                  None  ...     1
3734    ahorsewithnoname28  ...     1
3735    knownothingwiseguy  ...     1

[3736 rows x 3 columns]
___________NOT DEPRESSED_____________
                    author  ... label
0               dorkymcgee  ...     0
1              treesbloom2  ...     0
2                   HinaPF  ...     0
3                   rtfg21  ...     0
4          SomeoneAlive123  ...     0
...                    ...  ...   ...
3511  ProfessionalCoat3609  ...     0
3512           Horizon2910  ...     0
3513      DefconOozeScarab  ...     0
3514                 svrae  ...     0
3515             NinjaTr33  .

In [ ]:
df = pd.concat([depressed_df,nondepressed_df]) # merge the datasets
df = df.sample(frac=1) #shuffle

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7252 entries, 373 to 3374
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   author  6876 non-null   object
 1   post    7252 non-null   object
 2   label   7252 non-null   int64 
dtypes: int64(1), object(2)
memory usage: 226.6+ KB


In [ ]:
print(df.groupby('label').label.count())
print("Total: " , len(df) , " entries")

label
0    3516
1    3736
Name: label, dtype: int64
Total:  7252  entries


In [ ]:
df = df.reset_index(drop=True)
df

author  ... label
0                  Alt_SWR  ...     0
1            black_mariah1  ...     1
2              markjlast83  ...     0
3      15yearsoldblackmale  ...     0
4             niceraindrop  ...     0
...                    ...  ...   ...
7247       CreatureWarrior  ...     1
7248       YouGotToBeCrazy  ...     1
7249        SedativeCorpse  ...     0
7250  Full-Consequence-447  ...     0
7251          SameJudgment  ...     1

[7252 rows x 3 columns]

In [ ]:
from google.colab import drive
drive.mount("/content/drive")


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
df.to_csv('unfiltered_depression_reddit_data.csv')
!cp unfiltered_depression_reddit_data.csv "drive/My Drive/"